python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -173,15 +191,15 @@ def test_sequence_boolean(seq):
     assert arr.to_pylist() == expected
 
 
-@parametrize_with_iterable_types
+@parametrize_with_sequence_types
```

Comment:
Tests that require being able to contain duplicated values (for example this tests has `expected = [np.bool_(True), None, np.bool_(False), None]` with `None` twice) can't work against all iterables because some iterable types don't allow duplicates (for example `set`)